packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -134,7 +134,12 @@ export class ProfilesCache {
             return validationResult;
         }
 
-        validationResult.port = Number(url.port);
+        if (url.protocol.includes("https") && url.port === "") {
```

Comment:
I would propose to add some unit tests for this function to cover this 'weird' behaviour.

What do you think about it? The tests will be pretty easy to write, since this function is pure.

And second point: maybe you can extract these magic numbers and strings into some understandable constants? Because, now, I understand, why these constants are here, but eventually, we will forget about it and why we need such a weird `443` and `https` here.
Does it make sense?

Reply:
Maybe we can check for the port value equal to 0?

Because now you are checking number for equality with empty string... Why?...

I think unit tests may cover such things..

Reply:
I will add some unit tests. I am checking against an empty string because that's what is stored in url.port if the user doesn't enter anything. I'll check what happens if they enter 0.

Reply:
OMG, they store empty string in number field?.... I mean, port can be only number, as far as I know...

Reply:
wouldn't it be better to just check for `!url.port` which would cover null and undefined?